options/options_test.go

```diff
@@ -550,6 +551,168 @@ func TestParseAndSetOptions(t *testing.T) {
 	})
 }
 
+type configTester struct {
+	description  string
+	yamlBytes    []byte
+	expectedOpts *ToolOptions
+	outcome      int
+}
+
+func runConfigFileTestCases(testCases []configTester) {
+	configFilePath := "./test-config.yaml"
+	args := []string{"--config", configFilePath}
+	defer os.Remove(configFilePath)
+
+	for _, testCase := range testCases {
+		if err := ioutil.WriteFile(configFilePath, testCase.yamlBytes, 0644); err != nil {
+			So(err, ShouldBeNil)
+		}
+		opts := New("test", "", "", "", false, EnabledOptions{true, true, true, true})
+		err := opts.ParseConfigFile(args)
+
+		var assertion func()
+		if testCase.outcome == ShouldSucceed {
+			assertion = func() {
+				So(err, ShouldBeNil)
+				So(opts.Auth.Password, ShouldEqual, testCase.expectedOpts.Auth.Password)
+				So(opts.URI.ConnectionString, ShouldEqual, testCase.expectedOpts.URI.ConnectionString)
+				So(opts.SSL.SSLPEMKeyPassword, ShouldEqual, testCase.expectedOpts.SSL.SSLPEMKeyPassword)
+			}
+		} else {
+			assertion = func() {
+				So(err, ShouldNotBeNil)
+			}
+		}
+
+		Convey(testCase.description, func() {
+			assertion()
+		})
+	}
+}
+
+func createExpectedOpts(pw string, uri string, ssl string) *ToolOptions {
+	opts := New("test", "", "", "", false, EnabledOptions{true, true, true, true})
+	opts.Auth.Password = pw
+	opts.URI.ConnectionString = uri
+	opts.SSL.SSLPEMKeyPassword = ssl
+	return opts
+}
+
+func TestParseConfigFile(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.UnitTestType)
+
+	Convey("should error with no config file specified", t, func() {
+		opts := New("test", "", "", "", false, EnabledOptions{})
+
+		// --config at beginning of args list
+		args := []string{"--config", "--database", "myDB"}
+		So(opts.ParseConfigFile(args), ShouldNotBeNil)
+
+		// --config at end of args list
+		args = []string{"--database", "myDB", "--config"}
+		So(opts.ParseConfigFile(args), ShouldNotBeNil)
+
+		// --config= at beginning of args list
+		args = []string{"--config=", "--database", "myDB"}
+		So(opts.ParseConfigFile(args), ShouldNotBeNil)
+
+		// --config= at end of args list
+		args = []string{"--database", "myDB", "--config="}
+		So(opts.ParseConfigFile(args), ShouldNotBeNil)
+	})
+
+	Convey("should error with non-existent config file specified", t, func() {
```

Comment:
I think the different positions here are overkill to just test a non-existent file.